docs/content/oauth-oidc.md

```diff
@@ -0,0 +1,132 @@
+---
+title: OAuth2 and OpenID Connect
+kind: misc
+weight: 5
+---
+
+In order to make policy decisions we commonly need to know the identity of the caller. Traditionally this was often done by providing a user or client identifier along with the request, and use that identifier to look up further information like user details or permissions from a remote data source. While this model works fairly well for many applications, it scales poorly in distributed systems such as microservice environments. If multiple services are involved serving a single request from a user, having them all ask an external service or database for information about the user would effectively mean doubling the number of total requests sent, and all in order to retrieve the same data for each service!
+
+## JSON Web Tokens
+
+To deal with this, OAuth2 established a basic protocol for obtaining _access tokens_, which would be issued by the authorization server and passed around between services in order to identify and authorize the caller. Using opaque tokens however poses many of the same problems as those faced when passing around user identifiers - rather than asking a database for more information the authorization server would instead have to be consulted whenever additional data was needed. While the OAuth2 specification leaves it to the implementation to decide on the format of the tokens, JSON Web Tokens (JWTs) have emerged as the most common one. Using JWTs as tokens include a number of benefits.
+
+* Self-contained - JWTs contain most, or all, of the details needed to identify the caller.
+* Signed - JWTs are signed by the issuer, meaning their authenticity can be verified by the receiver.
+* Immutable - a JWT can't be changed anywhere in the request chain without invalidating the signature.
```

Comment:
Since you're taking some time to introduce JWTs, and how we got here; would you think it made sense to include Cons?

I suppose I could see this go either way, make the picture more complete with disadvantages; or reduce the introduction. As in, someone coming to the page to read this might already know what OIDC and JWT is, and might already use it, and now they want to use it with OPA. So the things they'd like to know might be

a. how verify a token (below, how to get the JWKS into OPA)
b. how to check/use its claims

and perhaps
c. how to test policies using JWTs.

What do you think? 😃